hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaUtil.java

```diff
@@ -75,7 +79,18 @@ public static Schema convert(List<FieldSchema> fieldSchemas, boolean autoConvert
       typeInfos.add(TypeInfoUtils.getTypeInfoFromTypeString(col.getType()));
       comments.add(col.getComment());
     }
-    return HiveSchemaConverter.convert(names, typeInfos, comments, autoConvert);
+    Set<Integer> identifierFieldIdSet = identifierFieldIdSetOrThrow(names, identifierFieldSet);
+    return HiveSchemaConverter.convert(names, typeInfos, comments, autoConvert, identifierFieldIdSet);
+  }
+
+  public static Set<Integer> identifierFieldIdSetOrThrow(List<String> names, Set<String> identifierFieldSet) {
+    return identifierFieldSet.stream()
+        .map(field -> {
+          int index = names.indexOf(field);
```

Comment:
Field IDs are not indexes, they are the integer identifiers embedded in an Iceberg schema.